src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/communitypage/SalesPitchDialog.java

```diff
@@ -1,37 +1,70 @@
 package com.wikia.webdriver.pageobjectsfactory.pageobject.communitypage;
 
+import com.wikia.webdriver.common.logging.PageObjectLogging;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
-
-import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.TimeoutException;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 
 public class SalesPitchDialog extends WikiBasePageObject {
 
-  private By salesPitchDialog = By.cssSelector("#CommunityPageBenefitsModal");
+  @FindBy(css = "#CommunityPageBenefitsModal")
+  private WebElement salesPitchDialog;
 
   @FindBy(css = "#CommunityPageBenefitsModal .community-page-entry-point-button")
-  private WebElement entryPointButton;
+  private WebElement helpOutButton;
+
+  @FindBy(css = ".community-page-benefits-image")
+  private WebElement image;
 
-  public boolean isVisible() {
-    wait.forElementVisible(salesPitchDialog);
+  @FindBy(css = ".community-page-benefits-content")
+  private WebElement dialogContent;
 
-    return true;
+  public boolean isDialogVisible() {
+    try {
+      wait.forElementVisible(salesPitchDialog, 5);
+      return true;
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.logInfo("Dialog is not visible");
+      return false;
+    } catch (TimeoutException e) {
+      PageObjectLogging.logInfo(e.getMessage());
+      return false;
+    }
   }
 
-  public boolean isNotVisible() {
-    wait.forElementNotVisible(salesPitchDialog);
+  public SpecialCommunity clickHelpOutButton() {
+    try {
+      wait.forElementVisible(helpOutButton);
+      helpOutButton.click();
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.logInfo("Help out button is not visible");
+    }
 
-    return true;
+    return new SpecialCommunity();
   }
 
-  public boolean clickEntryPointButton() {
-    isVisible();
+  public SpecialCommunity clickDialogImage() {
+    try {
+      wait.forElementVisible(image);
+      image.click();
+    } catch (NoSuchElementException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)